backend/RayTracing/Classes/Models/Ray.py

```diff
@@ -3,9 +3,21 @@
 
 class Ray:
 
-    def __init__(self, startp=Vector(0, 0, 0), direction=Vector(0, 0, 0)):
+    def __init__(self, startp=Vector(0, 0, 0), directionPoint=Vector(0, 0, 0)):
         self.startPoint = startp
-        self.direction = direction
+        self.direction = directionPoint.sub(startp)
```

Comment:
This change created a lot of bugs on the way current Rays work.
I tested it a bit using the old Rays and the new Rays and it creates a bug on reflection and I am guessing refraction as well. 
The reflectionRay direction and refractionRay direction are calculated in the RayTracer, so of course with the new Rays it does not work.

If we want to keep this change, I am guessing we should also have a function to normalize the Direction of the Ray from the RayTracer.

Reply:
Thank you for the explanation and the advice! Very good point. For the arbitrary camera position we need this calculation. Maybe we should change the RayTracer.